extensions/aws/s3/S3Wrapper.cpp

```diff
@@ -30,46 +37,253 @@ namespace minifi {
 namespace aws {
 namespace s3 {
 
-minifi::utils::optional<Aws::S3::Model::PutObjectResult> S3Wrapper::sendPutObjectRequest(const Aws::S3::Model::PutObjectRequest& request) {
-  Aws::S3::S3Client s3_client(credentials_, client_config_);
-  auto outcome = s3_client.PutObject(request);
+void HeadObjectResult::setFilePaths(const std::string& key) {
+  absolute_path = key;
+  std::tie(path, filename) = minifi::utils::file::FileUtils::split_path(key, true /*force_posix*/);
+}
+
+S3Wrapper::S3Wrapper() : request_sender_(minifi::utils::make_unique<S3ClientRequestSender>()) {
+}
+
+S3Wrapper::S3Wrapper(std::unique_ptr<S3RequestSender> request_sender) : request_sender_(std::move(request_sender)) {
```

Comment:
Since `unique_ptr` is move-only and we're just storing it, we can just take an rvalue reference for the same effect with one less move operation, assuming no elision.

Reply:
I am not sure how would you go about it, could you please elaborate? How would you initialize the `unique_ptr` with a `S3RequestSender&&`?

Reply:
```suggestion
S3Wrapper::S3Wrapper(std::unique_ptr<S3RequestSender>&& request_sender) : request_sender_(std::move(request_sender)) {
```

Reply:
Thanks, sorry I misinterpreted your comment at first, updated in 4001582e1efc9d5c697854e8b4477eb393d6e2f3